horovod/mxnet/mpi_ops.h

```diff
@@ -33,16 +33,18 @@ using namespace horovod::common;
 typedef ::mxnet::NDArray NDArray;
 typedef ::mxnet::Engine::CallbackOnComplete CallbackOnComplete;
 typedef Request::RequestType OperationType;
+typedef std::shared_ptr<MXTensor<NDArray>> MXTensorSharedPtr;
```

Comment:
hvd_cpu_buffer is [created](https://github.com/horovod/horovod/pull/1304/files#diff-79b78539e15db1d75d73029490eed708R190) as a shared ptr of MXTemporaryBuffer. Here we define a type of shared ptr for MXTensor. Do they need to be consistent? Can we use either MXTensor or MXTemporaryBuffer?

Reply:
I used `MXTemporaryBuffer` in `PushHorovodOperationCudaOnCPU()` because it has the destructor method to delete the tensor when not in use. In the above line, I have used `MXTensor` to make it more generaliz(can be use in future scenarios).